src/index.ts

```diff
@@ -42,31 +49,33 @@ async function getTotalVotersBalance(voteData: VotingData): Promise<BigNumber> {
   return totalVotersBalance;
 }
 
-export async function main(): Promise<Map<string, BigNumber>> {
+export async function calculateRewardsAmounts(
+  TotalDistributionAmount: BigNumber
+): Promise<Rewards> {
   const voteData = await fetchVoteData(web3, voteId);
   // snapshot blockNumber of DAO proposal vote
   const blockNumber = voteData.snapshotBlockNumber;
   // fetch total amount of SarcoVR held by voters for a given voteId
   const totalVoteBalance = await getTotalVotersBalance(voteData);
 
-  // construct mapping of voters' addresses and rewards to be allocated
-  const distributionMapping = new Map<string, BigNumber>();
+  // construct object array of voters' addresses and rewards to be allocated
+  const rewardsObject: Rewards = [];
```

Comment:
This is confusing. Is `Rewards` an array?

Ah I just noticed the type definition on line 14 above. Ehh no I'd much rather you explicitly used `Rewards[]` to type your array variables instead, and removed that indirection.